flake/dev/flake.nix

```
@@ -0,0 +1,51 @@
+{
+  description = ''
+    Private inputs for development purposes. These are used by the top level
+    flake in the `dev` partition, but do not appear in consumers' lock files.
+  '';
+
+  inputs = {
+    # This input is prefixed with 'dev-' to avoid shadowing the public flake's
+    # 'nixpkgs' input, allowing other dev inputs to follow it.
```

This wording is awkward; it implies we need to avoid shadowing in order to follow it, but that's not the case.

Shadowing and following are two unrelated things.

1. We need to define a nixpkgs input so that we can follow it when creating the lockfile 
2. We need to use a different name to avoid shadowing while evaluating the partition submodule


```suggestion
    # We need to define a nixpkgs input so that other inputs can follow it.
    # It is prefixed with 'dev-' to avoid shadowing the public flake's
    # 'nixpkgs' input, which could affect local testing with --override-input.
```